src/core/trtserver.h

```diff
@@ -465,13 +466,103 @@ TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_TraceManagerNew(
 TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_TraceManagerDelete(
     TRTSERVER_TraceManager* trace_manager);
 
+/// TRTSERVER_InferenceRequestOptions
+///
+/// Object representing the request meta-data needed for an inference.
+///
+
+/// Create a new inference request options object.
+/// \param request_options Returns the new request options object.
+/// \param model_name The name of the model that the inference request is for.
+/// \param model_version The version of the model that the inference
+/// request is for, or -1 to select the latest (highest numbered) version.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_InferenceRequestOptionsNew(
+    TRTSERVER_InferenceRequestOptions** request_options, const char* model_name,
+    int64_t model_version);
+
+/// Set the ID for the request in a request options. The response of the request
+/// will contain the same ID. The request sender can use the ID to correlate
+/// the response to corresponding request if needed.
+/// \param request_options The request options object.
+/// \param id The ID.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_InferenceRequestOptionsSetId(
+    TRTSERVER_InferenceRequestOptions* request_options, uint64_t id);
+
+/// Set the flag associated with the request in a request options. 'flags'
+/// should holds a bitwise-or of all flag values. \param request_options The
+/// request options object. \param flags The flags. \return a TRTSERVER_Error
+/// indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_InferenceRequestOptionsSetFlags(
+    TRTSERVER_InferenceRequestOptions* request_options, uint32_t flags);
```

Comment:
Need to define an enumeration for the flags (see api.proto). Similar naming, etc. as we have with TRTSERVER_Model_Control_Mode except that must use explicit power-of-2 enum values.